pandas/io/parsers.py

```diff
@@ -1708,7 +1708,9 @@ def _convert_to_ndarrays(
         result = {}
         for c, values in dct.items():
             conv_f = None if converters is None else converters.get(c, None)
-            if isinstance(dtypes, dict):
+            if values.dtype != object:
```

Comment:
this is very odd to do as we already have a path for a single dtype, what are you trying to do here?

Reply:
After loading values from csv we have dictionary with column names and numpy array for each column with `dtype=object`. Then we change values that are suppose to be datetime (`'b'` in example). After that we want to change types of the rest of columns, that is those that have `dtype=object`. In that line we're skiping columns that already have dtype set.